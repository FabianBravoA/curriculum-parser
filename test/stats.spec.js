const { computeCourseStats } = require('../lib/stats');


describe('stats.computeCourseStats', () => {
  it('should be a function', () => expect(typeof computeCourseStats).toBe('function'));

  it('should compute course duration based on parts duration', () => {
    const { stats } = computeCourseStats({
      syllabus: {
        foo: {
          parts: {
            '01-intro': { duration: 10 },
          },
        },
        bar: {
          parts: {
            '00-opening': { duration: 5 },
            '00-closing': { duration: 5 },
          },
        },
      },
    });

    expect(stats).toMatchSnapshot();
  });
});
