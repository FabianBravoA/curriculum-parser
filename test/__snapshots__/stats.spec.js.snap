// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stats.computeCourseStats should compute course duration based on parts duration 1`] = `
Object {
  "duration": 20,
  "durationString": "20min",
  "exerciseCount": 0,
  "partCount": 3,
  "unitCount": 2,
}
`;
